lib/src/util/posix/pid_file.cc

```diff
@@ -23,13 +22,16 @@ namespace Util {
 namespace fs = boost::filesystem;
 namespace lth_file = leatherman::file_util;
 
-const std::string PIDFile::FILENAME { "pxp-agent.pid" };
-
-PIDFile::PIDFile(const std::string& dir_path_)
-        : dir_path { dir_path_ },
-          file_path { dir_path + "/" + FILENAME },
+PIDFile::PIDFile(const std::string& file_path_)
+        : dir_path { fs::path(file_path_).parent_path().string() },
+          file_path { file_path_ },
           pidfile_fd {},
           cleanup_when_done { false } {
+    if (fs::exists(file_path) && !(fs::is_regular_file(file_path))) {
```

Comment:
Aren't we in trouble if the PID file already exists? Should this be a check for, "if the pidfile exists, terminate with an error"?